### PR TITLE
chore(eslint) - update eslint to require import type

### PR DESCRIPTION
**Pull request by devinea**
TBI https://github.com/SAP/open-ux-tools/issues/382

- update eslint to require `import type` for ts types instead of `import`
- disable `no-use-before-define` and use `@typescript-eslint/no-use-before-define` instead.
- Add missing lint:fix scripts for some packages
- Switch `.eslintrc` to consistently use off/warn/error instead of 0/1/2.
- remove `brace-style` as it conflicts with prettier hook and use prettier eslint instead
- replace `jsxBracketSameLine` (deprecated) with `bracketSameLine`
-  eslint `prettier/prettier` configure `endOfLine` to avoid errors in pipeline and across OSes

### CODE REVIEW

**Comment by devinea** [comment]
Sonar code smells
@ullasholla - Could you take a look at https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=374&id=SAP_open-ux-tools&open=AX9kxKbGvrM0-VuqnINS

@zdravko-georgiev - Could you take a look at https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=374&id=SAP_open-ux-tools&open=AX9kxKhFvrM0-VuqnINY

**Comment by zdravko-georgiev** [comment]
> Sonar code smells @ullasholla - Could you take a look at https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=374&id=SAP_open-ux-tools&open=AX9kxKbGvrM0-VuqnINS
> 
> @zdravko-georgiev - Could you take a look at https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=374&id=SAP_open-ux-tools&open=AX9kxKhFvrM0-VuqnINY

Hi @devinea, can we mark this as `won't fix`? It is by intention that the code looks like this. Unfortunately I can't mark it myself.

**Review by zdravko-georgiev — DISMISSED** [review]

**Comment by devinea** [comment]
> > Sonar code smells @ullasholla - Could you take a look at https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=374&id=SAP_open-ux-tools&open=AX9kxKbGvrM0-VuqnINS
> > @zdravko-georgiev - Could you take a look at https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=374&id=SAP_open-ux-tools&open=AX9kxKhFvrM0-VuqnINY
> 
> Hi @devinea, can we mark this as `won't fix`? It is by intention that the code looks like this. Unfortunately I can't mark it myself.

Thanks @zdravko-georgiev I've marked as won't fix.

**Comment by ullasholla** [comment]
> Sonar code smells @ullasholla - Could you take a look at https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=374&id=SAP_open-ux-tools&open=AX9kxKbGvrM0-VuqnINS
> 
@devinea, the fix turned out to be a bit involved and extensive. Hopefully, this should fix it: https://github.com/SAP/open-ux-tools/pull/374/commits/6287cae51574a044272e9eebcfdf3de1e7ed8649

**Review by zdravko-georgiev — APPROVED** [review]

**Comment by devinea** [comment]
@ullasholla Thanks for taking a look. Unfortunately the code smells are more or less still there.

**Comment by devinea** [comment]
Thanks @ullasholla

File: packages/store/src/data-access/filesystem.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-import { DataAccess, DataAccessConstructor } from '.';
-import { Logger } from '@sap-ux/logger';
+import type { DataAccess } from '.';
+import type { Logger } from '@sap-ux/logger';
 import { errorInstance, getFioriToolsDirectory } from '../utils';
-import { Entity } from '..';
-import { ServiceOptions } from '../types';
+import type { ServiceOptions } from '../types';
 import os from 'os';
+import type { Entity } from 'contants';
```
Comment (IainSAP) [inline packages/store/src/data-access/filesystem.ts:10]: `contants` ?

File: packages/store/src/data-access/filesystem.ts [hunk 2]
```diff
@@ -32,20 +28,14 @@ export const FilesystemStore: DataAccessConstructor<object> = class implements D
...
     }
 
-    public async read<Entity extends object>({
-        entityName,
-        id
-    }: {
-        entityName: string;
-        id: string;
-    }): Promise<undefined | Entity> {
+    public async read({ entityName, id }: { entityName: string; id: string }): Promise<undefined | E> {
```
Comment (IainSAP) [inline packages/store/src/data-access/filesystem.ts:31]: Removal of generic, is that expected?

**Review by IainSAP — CHANGES_REQUESTED** [review]
OK for FE, FF, odata-service. Some unexpected changes in `store`?

**Comment by IainSAP** [comment]
@devinea Can you link the associated TBI in the description please?

File: packages/store/src/data-access/filesystem.ts (hunk 2 above)
Comment (devinea) [reply packages/store/src/data-access/filesystem.ts:31]: Yeah this was showing up as a code smell in sonar.

File: packages/store/src/data-access/filesystem.ts (hunk 1 above)
Comment (devinea) [reply packages/store/src/data-access/filesystem.ts]: Fixed in https://github.com/SAP/open-ux-tools/pull/374/commits/b43b771dd66aa53af1cc90b47e0ae4ede92c55aa

**Comment by IainSAP** [comment]
> @devinea Can you link the associated TBI in the description please?

Can you add the value also - why are we doing this rather than what we are doing?

**Review by IainSAP — APPROVED** [review]
OK for FF, FE, odata-service. No unexpected code changes. Thx @devinea

**Review by kranthie-sap — APPROVED** [review]
eslint and prettier config changes look good to me. Thanks, @devinea.

**Review by tobiasqueck — APPROVED** [review]
I didn't see a functional change, only improvements of imports or and configurations. Since all build and test scripts run without issues, and I don't see any wrong change in your 73 files, I am ok with that change BUT I have to admit, then I have not reviewed every single changed line.

**Pull request merged**